### ISSUE TITLE
[Feature request] Allow plugins to listen when a tab is activated

### ISSUE THREAD

**Issue by solid-studio** [ethereum/remix-ide#2572]
As mentioned in https://github.com/ethereum/remix-plugin/issues/160

I was wondering if would be possible to allow plugins to know when the tab of the plugin is active.

**Comment by GrandSchtroumpf** [comment]
I've implemented the interface `isFocus` on HostPlugin (sidePanel, mainPanel). We can also emit an event `focused` when method `focus` is triggered.

Here is a simple implementation in remix-ide :

```typescript
export class Panel extends HostPlugin {
  plugins: Record<string, HTMLElement> = {}
  focused: string
  root: Element // The root html element of the HostPlugin
  constructor(name: string, root = document.body) {
    super({ name })
    this.root = root
  }
  // Add a View plugin into the DOM once it's activated
  addView(profile: Profile, view: HTMLElement) {
    this.plugins[profile.name] = view
    view.style.display = 'none'
    this.root.appendChild(view)
  }
  // Remove a View plugin from the DOM once it's deactivated
  removeView(profile: Profile) {
    this.root.removeChild(this.plugins[profile.name])
    if (this.focused === profile.name) delete this.focused
    delete this.plugins[profile.name]
  }
  // Set focus on one plugin
  focus(name: string) {
    if (this.plugins[name]) {
      if (this.plugins[this.focused]) this.plugins[this.focused].style.display = 'none'
      this.plugins[name].style.display = 'block'
      this.focused = name
      this.emit('focused', name)
    }
  }
  // Check if a plugin is focus
  isFocus(name: string) {
    return name === this.focused
  }
}

const sidePanel = new Panel('sidePanel' document.querySelector('#sidebar'))
```

@solid-studio in your plugin you should be able to do :
```typescript
client.on('sidePanel', 'focused', name => {
  if (name === 'your-plugin') ...
})
```

**Comment by yann300** [comment]
> I've implemented the interface isFocus 

you mean it is already there?

**Comment by GrandSchtroumpf** [comment]
It's an abstract method of the HostPlugin in version 0.2.0, but it can be implemented in v 0.1.x to be available sooner.

**Comment by LianaHus** [comment]
https://github.com/ethereum/remix-ide/issues/2804
